### ISSUE TITLE
STP fails on nested bvudivs

### ISSUE THREAD

**Issue by hriener** [agra-uni-bremen/metaSMT#31]
```
BOOST_AUTO_TEST_CASE(stp_error)
{
  const unsigned w = 8u;
  bitvector v1 = new_bitvector(w);
  bitvector v2 = new_bitvector(w);
  bitvector v3 = new_bitvector(w);
  bitvector v4 = new_bitvector(w);
  bitvector v5 = new_bitvector(w);
  bitvector v6 = new_bitvector(w);

  ContextType::result_type c = evaluate( ctx, equal(bvudiv(bvudiv(v2,v3),bvudiv(bvudiv(v1,v2),bvudiv(v1,v3))),bvand(v6,bvand(v4,v5))) );
  assertion( ctx, Not( equal( Ite( Not( c ), logic::True, logic::False ), logic::False ) ) );
  BOOST_REQUIRE( solve(ctx) );
}


./tests/direct_STP -t QF_BV/stp_error
Running 1 test case...
Fatal Error: BVConstEvaluator:division by zero error
direct_STP: /home/development-vm/development/metaSMT/build/build/stp-git/lib/AST/ASTmisc.cpp:272: void stp::FatalError(const char*): Assertion `false' failed.
unknown location(0): fatal error in "stp_error": signal: SIGABRT (application abort requested)
/home/development-vm/development/metaSMT/tests/test_QF_BV.cpp(1963): last checkpoint

*** 1 failure detected in test suite "direct_STP"
```

**Comment by finnhaedicke** [comment]
Will you look into this yourself?

**Comment by hriener** [comment]
Yep, if that is okay for you.

metaSMT seems to be in a pretty good shape. After running a huge test suite, I currently see only a few regressions.

**Comment by finnhaedicke** [comment]
Sure, go for it. You fixed the issue yesterday in the time I needed to setup a test case. ;-)

**Comment by hriener** [comment]
Constant setup time will disappear in an asymptotic analysis. :smile:

**Comment by hriener** [comment]
fixed in 8b137fbc65efff09fb32aa78e837128436278653
